crates/store/re_log_encoding/src/decoder/stream.rs

```diff
@@ -67,11 +129,86 @@ enum State {
     /// Compression is only applied to individual `ArrowMsg`s, instead of the entire stream.
     Message(crate::codec::file::MessageHeader),
 
-    /// Stop reading
+    /// Stop reading.
     Aborted,
 }
 
-impl StreamDecoder {
+impl<T: FileEncoded> StreamDecoder<T> {
+    /// Instantiates a new lazy decoding iterator on top of the given buffered reader.
+    ///
+    /// This does not perform any IO until the returned iterator is polled. I.e. this will not
+    /// fail if the reader doesn't even contain valid RRD data.
+    ///
+    /// This takes a `BufRead` instead of a `Read` because:
+    /// * This guarantees this will never run on non-buffered input.
+    /// * This lets the end-user in control of the buffering, which prevents unfortunately stacked
+    ///   buffers (and thus exploding memory usage and copies).
+    ///
+    /// See also [`Self::decode_lazy_with_opts`].
+    pub fn decode_lazy<R: std::io::BufRead>(reader: R) -> StreamDecoderIterator<T, R> {
+        let wait_for_eos = false;
+        Self::decode_lazy_with_opts(reader, wait_for_eos)
+    }
+
+    /// Same as [`Self::decode_lazy`], with extra options.
+    ///
+    /// * `wait_for_eos`: if true, the decoder will always wait for an end-of-stream marker before
+    ///   calling it a day, even if the underlying reader has nothing to yield.
```

Comment:
Explain _when_ this makes sense. I'm pretty sure the only case that makes sense is when using a RetryableFileReader, right (not that it hurts necessarily otherwise, but why would you though?)?